framework/src/actions/ActionWarehouse.C

```diff
@@ -349,8 +349,12 @@ ActionWarehouse::printInputFile(std::ostream & out)
   for (std::vector<Action* >::iterator i = ordered_actions.begin();
        i != ordered_actions.end();
        ++i)
-   {
-    std::string name ((*i)->name());
+  {
+    std::string name;
+    if ((*i)->isParamValid("parser_syntax"))
+      name = (*i)->getParam<std::string>("parser_syntax");
+    else
+      name = (*i)->name();
```

Comment:
I will look into this. I fixed this up a few weeks ago, but I will check that it still works.